src/ResponseSerializerInterface.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Yiisoft\Yii\Rest;
+
+use Psr\Http\Message\ResponseInterface;
+
+interface ResponseSerializerInterface
+{
+    // TODO may be $code is redundant?
+    public function serialize(int $code, $data): ResponseInterface;
```

Comment:
Serialization usually means getting a string. In this case we're getting response. Would `createResponse(int $code, $data)` be more appropriate?

Reply:
How about to rename class from serializer to assembler and method serialize to assemble?

Reply:
An object that assembles another object is usually called a factory.